### PR TITLE
Upgrade go to v1.23

### PR DESCRIPTION
**Pull request by Ex4amp1e**
## Description
Upgrade go version

Proof of concept of working CI: https://github.com/Ex4amp1e/sdk/pull/2 (using updated workflows)

## Issue link
https://github.com/networkservicemesh/deployments-k8s/issues/12294

## How Has This Been Tested?

- [ ] Added unit testing to cover
- [ ] Tested manually
- [ ] Tested by integration testing
- [x] Have not tested

## Types of changes

- [ ] Bug fix
- [x] New functionality
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

**Comment by denis-tingaikin** [comment]
@Ex4amp1e Could we also use changes from https://github.com/networkservicemesh/sdk/pull/1548?

File: .golangci.yml [hunk 1]
```diff
@@ -57,81 +80,51 @@ linters-settings:
   gosec:
     excludes:
       - G115
-      - G301 # Fixable
-      - G302 # Fixable
-      - G306 # Fixable
+      - G204
```
Comment (denis-tingaikin) [inline .golangci.yml:83]: Why are we excluding `G204`?

Comment (Ex4amp1e) [reply .golangci.yml:83]: G204 was included in exclude-rules list before, but on the current linter version we are getting too many new gosec errors, so decided to put them into one place- general config, not individually

Comment (denis-tingaikin) [reply .golangci.yml:83]: OK, add a todo and refer to the issue where we plan to fix these issues.

File: .golangci.yml [hunk 2]
```diff
@@ -293,11 +246,16 @@ issues:
...
+        - unused
+    - path: pkg/networkservice/common/switchcase/common_test.go
+      linters:
+        - revive
+    - path: pkg/tools/opa/policies.go
       linters:
-        - interfacer
-      text: "`rawSrv` can be `github.com/networkservicemesh/sdk/pkg/networkservice/common/monitor.EventConsumer`"
+        - revive
+    # TODO - remove deprecated opentelemetry functions
```
Comment (denis-tingaikin) [inline .golangci.yml:258]: Please add a ticket to refer here. Thanks.

File: .golangci.yml (hunk 1 above)
Comment (Ex4amp1e) [reply .golangci.yml:84]: Done

File: .golangci.yml (hunk 2 above)
Comment (Ex4amp1e) [reply .golangci.yml]: Done

**Review by denis-tingaikin — APPROVED** [review]
Looks acceptable. 

@NikitaSkrynnik Could you also have a look?

**Pull request merged**